meson.build

```diff
@@ -0,0 +1,123 @@
+project('libtexpdf', 'c', version : '2.4.6', default_options : ['c_std=gnu11'])
+
+pkg = import('pkgconfig')
+
+compiler = meson.get_compiler('c')
+
+conf = configuration_data()
+conf.set('version', meson.project_version())
+
+check_headers = ['dlfcn.h', 'inttypes.h', 'stdint.h', 'stdlib.h', 'string.h', 'sys/stat.h', 'sys/types.h', 'sys/wait.h', 'unistd.h']
+foreach header_name : check_headers
+    if compiler.has_header(header_name)
+        conf.set('HAVE_' + header_name.underscorify().to_upper(), 1)
```

Comment:
This is a port from the existing code, but it's not really necessary -- autotools sometimes checks for things without you asking ;) and the real question is, what does the code itself check for?
- sys/types.h
- sys/wait.h
- sys/stat.h
- stdlib.h
- stdbool.h

Reply:
Only sometimes? ;-) I've never been able to figure out how to stop it from doing hundreds of useless checks. You would think it'd detect a modern Linux kernel and then just give up on all that but nope.

Reply:
;)

Reply:
> This is a port from the existing code, but it's not really necessary -- autotools sometimes checks for things without you asking ;) and the real question is, what does the code itself check for?

I think too all these checks are pretty useless and I feel these are a legacy from the the autotools. The autotools go to a great length to support obscure Unix systems most of which are no longer in use. To these purpose autotools makes a lot of checks and tends to introduce a lot of accidental engineering complexity into the project itself.

My idea was to match at first the autotools using a Meson config and later, if we want to, remove some of these unneeded cruft. I just omitted from Meson all the checks that are done by the configure script but are not actually used in the code.

Reply:
I agree with you but, for example, HAVE_SYS_TYPES_H is actually used in the code so I added it in the meson configuration. The same goes for the others.

Reply:
There are two that you're checking for which I cannot find in the code, (there's one you're not checking for that is used).

Reply:
@eli-schwartz, ok sorry, I did not clearly understood at the beginning. I have normally fixed this problem, please have a look.